integr8ly/integreatly-operator
# INTLY-2833: Install Fuse on OpenShift templates and imagestreams

## What
Adds Fuse on OpenShift (templates and imagestreams) installation to the operator. 
- [x] Add reconciler for Fuse on OpenShift
   - This should install imagestreams and templates from this [repo](https://github.com/jboss-fuse/application-templates/)
   - **NOTE**: Some imagestreams/templates are already managed by the sample cluster operator. The names of the imagestreams and templates are added to the `SkippedImagestreams` and `SkippedTemplates` property in the sample cluster config CR. This enables us to update them without the sample cluster operator reconciling them again.
- [x] Bump operator version
- [x] Add Fuse on OpenShift to managed and workshop product types
- [x] Add tests for reconciler

This PR also contains changes to update `openshift/api` package to release-4.3. This is required to import the sample cluster operator types.

The console cluster and namespace templates for OS4 are not available in a GA release yet. Therefore, this PR currently points to `master` of the [application templates](https://github.com/jboss-fuse/application-templates) repo.

## Verification Steps
1. Install the integreatly operator and create an installation CR
    - Install using an operator source pointing to `jameelb` app registry or run locally
2. **Verify**: Fuse on OpenShift imagestreams and templates are specified to be skipped by the sample cluster config operator
    - Search for `Config` custom resource of kind `samples.operator.openshift.io/v1`
    - Select `cluster` and go to the YAML definition.
    - The names of the imagestreams should be listed in the `spec.skippedImagestreams`
    - The names of the templates should be listed in the `spec.skippedTemplates`
3. **Verify**: Fuse on OpenShift imagestreams should be installed successfully
    - Go to `openshift` namespace
    - Image streams listed in this [file](https://github.com/jboss-fuse/application-templates/blob/master/fis-image-streams.json) should be available with a label `integreatly=true`
4. **Verify**: Fuse on OpenShift templates should be installed successfully
    - Go to `openshift` namespace
    - Quickstart templates listed [here](https://github.com/jboss-fuse/application-templates/tree/master/quickstarts) should be available with a label `integreatly=true`
    - Console [cluster](https://github.com/jboss-fuse/application-templates/blob/master/fuse-console-cluster-os4.json) and [namespace](https://github.com/jboss-fuse/application-templates/blob/master/fuse-console-namespace-os4.json) templates should be available with a label `integreatly=true`
    - [Apicurito](https://github.com/jboss-fuse/application-templates/blob/master/fuse-apicurito.yml) template should be available with a label `integreatly=true`
5. **Verify**: Product status in the Installation CR status should be updated with phase complete and the product version
6. **Verify**: All resources created by the installation should be deleted if the installation CR was removed.
